klpbuild/klplib/bugzilla.py

```diff
@@ -5,26 +5,172 @@
+            component="Kernel Live Patches",
+            assigned_to="kernel-lp")
+
+    query["ids_only"] = True
+    ids = [b.id for b in __bzapi.query(query)]
+    bugs = __bzapi.getbugs(ids)
+
+    deps_ids = [b.depends_on[0] for b in bugs if len(b.depends_on) > 0]
+    deps_fields = ["status", "resolution", "assigned_to", "whiteboard"]
+    deps = {d.id:d for d in __bzapi.getbugs(deps_ids,include_fields=deps_fields)}
+
+    return bugs, deps
+
+
+@__check_is_connected
+def get_bug(bsc):
+    """
+    Fetch a bug's data indicated by the given bug id.
+
+    Args:
+        bsc (str/int): bug id. Must be a numerical value.
+
+    Returns:
+        bug object.
+    """
+    if isinstance(bsc, str) and not bsc.isnumeric():
+        return None
+
+    return __bzapi.getbug(bsc)
+
+
+def get_bug_classification(bug):
+    """
+    Return the difficulty score assigned by Nicolai in the comment section.
+    """
+    rating = ["trivial", "medium", "complex"]
+    comments = []
+    i = 0
+
+    while i < 5:
+        try:
+            comments = bug.getcomments()
+            break
+        except:
+           # There's a max number of allowed simultaneous requests...
+            time.sleep(5)
+            i += 1
+
+    for c in comments:
+        if "nstange" in c["creator"]:
+            return w if any((w:=i) in c["text"] for i in rating) else "unknown"
 
-    return bzapi.getbug(bsc)
+    return "None"
 
 
-def get_bug_title(bsc):
+def get_bug_dep(bug, deps=None):
     """
-    The bug description usually is after the "kernel live patch" message. Return a fixes message
-    if the livepatch name is not an ID from a bug on bugzilla.
+    Return the corresponding dependency to the given bug.
+    To speed up the lookup it allows passing a precalculated
+    list will all the dependencies for all bugs, instead
+    of fetching them from bugzilla one by one.
     """
-    if not bsc.isnumeric():
+
+    # XXX: Support returning more than one dependency?
+
+    d = bug.depends_on
+    if not d:
+        return None
+
+    if deps:
+        return deps[d[0]]
+
+    return get_bug(d[0])
+
+
+def get_bug_cvss(bug):
+    if bug is None:
+        return "None"
+
+    raw = bug.whiteboard.split(':')
+    return raw[3] if len(raw) >= 4 else "None"
+
+
+def get_bug_summary(bug):
+    return bug.summary.split(':')
+
+
+def get_bug_cve(bug):
+    summary = get_bug_summary(bug)
+    if len(summary) < 2:
+        return ""
+
+    return summary[1][5:].strip()
+
+
+def get_bug_subsys(bug):
+    summary = get_bug_summary(bug)
+    if len(summary) < 3:
+        return "Unknown"
+
+    return summary[3][:40].strip()
+
+
+def get_bug_prio(bug):
+    return bug.priority[5:]
+
+
+def is_bug_dropped(bug):
+    return bug.resolution and bug.resolution in {"INVALID", "WONTFIX", "DUPLICATED"}
+
+
+def get_bug_data(bug, dep):
+    return (get_bug_cve(bug), get_bug_subsys(bug), get_bug_cvss(dep),
+            get_bug_classification(dep), get_bug_prio(bug))
```

Comment:
```suggestion
            get_bug_classification(bug), get_bug_prio(bug))
```
Shouldn't we read the classification from the livepatching bug?

Reply:
Indeed, good catch. Thanks!